### ISSUE TITLE
Unexpected label <num>

### ISSUE THREAD

**Issue by aurooj** [jay-mahadeokar/deeplab-public-ver2#1]
Hi,

I am facing an issue which makes no sense to me. When I try to finetune deeplab on my dataset which has only 2 classes including background. I keep on getting this error. I tried changing ignore_label value but it doesn't help, so I kept that to 255. 

Any hints what causes this error to pop up? 

Also, in deeplab's older version, we could add weight_source file to loss layer if our classes are unbalanced. But I checked into updated softmax_loss_layer.cpp, this parameter is not being handled anymore. Any specific reasons for omitting it? Also, what is the work around to penalize for class which has few pixels?

Any help will be highly appreciated!
Thanks.

**Comment by aurooj** [comment]
Okay, so I tried printing out the labels it takes from my image and what goes inside seg_accuracy_layer.cpp and figured out that, for some reason, it changes my actual label value which is 1 to 38 during interpolation. 

Can anyone please explain me what can be the possible reason? I am sorry if I am making some obvious mistake here, but I will appreciate if anyone points it out to me.

Thanks in advance!

**Comment by ajdroid** [comment]
Did you figure this out?

**Comment by ajdroid** [comment]
Did you figure this out, @aurooj ?

**Comment by aurooj** [comment]
I couldn't resolve this issue. But since the issue was just in SegAccuracy layer, I replaced this layer with Accuracy layer, and then everything worked fine for me.

**Comment by ajdroid** [comment]
Hmm, that works but I'm still curious as to what causes this.

**Comment by aurooj** [comment]
Yes, I think it should be a bug since many people are reporting this issue.
